platform/files.go

```diff
@@ -85,17 +86,67 @@ type BuildMetadata struct {
 	PlatformAPI                 *api.Version             `toml:"-" json:"-"`
 }
 
+// DecodeBuildMetadataTOML reads a metadata.toml file
+func DecodeBuildMetadataTOML(path string, platformAPI *api.Version, buildmd *BuildMetadata) error {
+	// decode the common bits
+	_, err := toml.DecodeFile(path, &buildmd)
+	if err != nil {
+		return err
+	}
+
+	buildmd.PlatformAPI = platformAPI
+
+	return nil
+}
+
 func (md *BuildMetadata) MarshalJSON() ([]byte, error) {
+	type BuildMetadataProcessSerializer struct {
+		Type             string   `toml:"type" json:"type"`
+		Command          string   `toml:"command" json:"command"`
+		Args             []string `toml:"args" json:"args"`
+		Direct           bool     `toml:"direct" json:"direct"`
+		Default          bool     `toml:"default,omitempty" json:"default,omitempty"`
+		BuildpackID      string   `toml:"buildpack-id" json:"buildpackID"`
+		WorkingDirectory string   `toml:"working-dir,omitempty" json:"working-dir,omitempty"`
+	}
```

Comment:
I think if we gave `launch.Process` a `MarshalJSON` method this logic could go away, no?

Reply:
Done! Good call. This might end up biting us a little bit when we need to serialize differently based on the platform API. We may need to pass that down as part of the `launch.Process` or re-read from env (if that is safe to do) during serialization.

Perhaps all our file structs would be better off if they always had `PlatformAPI string `toml: "-", json: "-"` so we can more easily implement `MarshalJSON` and `MarshalText`. I think the dynamic `buildpack.BOMEntry` in this same area could have also maybe been implemented similar to `launch.Process`